### PR TITLE
Trapping rain water in java

### LINKED ISSUE #5078: Trapping rainwater in Java
## 💥 Proposal

Given n non-negative integers representing an elevation map where the width of each bar is 1, compute how much water it can trap after raining.
I am a GSSOC participant and want to implement it in java.
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Amit366**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added code for Trapping rain water in java

### Checklist

- [ ] I've read the contribution guidelines.
- [ ] I've checked the issue list before deciding what to submit.
- [ ] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes: #5078

### CODE REVIEW

**Comment by Amit366** [comment]
@ankitaggarwal23 please review.

File: C/cp/Median_of_two_sorted_arrays.c [hunk 1]
```diff
@@ -1,108 +0,0 @@
...
-Enter sizes of both arrays:
-5 3
-Enter elements of first array:
-1 5 7 8 9
-Enter elements of second array:
-2 4 4
-Output:
-Median of two arrays is 4.50000
-
-*/
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c]: Why did you delete this file?

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Comment by Amit366** [comment]
@HarshCasper this is already merged

File: C/cp/Median_of_two_sorted_arrays.c [hunk 2]
```diff
@@ -1,3 +1,88 @@
+<<<<<<< HEAD
```
Comment (HarshCasper) [inline C/cp/Median_of_two_sorted_arrays.c:1]: You seem to have messed up with git

**Review by HarshCasper — CHANGES_REQUESTED** [review]

**Review by ankitaggarwal23 — COMMENTED** [review]
Don't push more than one code file in a single PR.

Do one thing start from fresh and don't directly create PR, when you are creating a PR there is one called make a draft create a draft one for review.
Need any help just DM us on discord and will guide you.
